make/common/modules/GensrcCommon.gmk

```diff
@@ -41,8 +41,8 @@ include $(TOPDIR)/make/ToolsJdk.gmk
 define SetupVersionProperties
   $(SUPPORT_OUTPUTDIR)/gensrc/$(MODULE)/$$(strip $2):
 	$$(call MakeTargetDir)
-	$(PRINTF) "jdk=$(VERSION_NUMBER)\nfull=$(VERSION_STRING)\nrelease=$(VERSION_SHORT)\n" \
-	    > $$@
+	$(PRINTF) "jdk=%s\nfull=%s\nrelease=%s\n" \
+	    $(VERSION_NUMBER) $(VERSION_STRING) $(VERSION_SHORT) > $$@
```

Comment:
Can't this just use echo like you did with the windows version elsewhere?

Reply:
No, there are embedded `\n` and that does not work with echo. I could split it into three consecutive lines but I don't think that would be a gain.

Reply:
Ah I see. So basically the only time we actually need `printf` as far as I can see is when there are embedded newlines in the output string? We could have used `printf` without `%s` format specifiers whenever we are substituting a variable.

Reply:
Yes, that, or that we need additional formatting, such as "%22s" where we want to align certain fields. Embedded `\n` can in theory be replaced with multiple calls to echo, but in the end it is a matter of judgement which is to prefer.